rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -225,15 +224,27 @@ public Blockchain getBlockchain() {
         return blockchain;
     }
 
+    public BlockFactory getBlockFactory() {
+        if (blockFactory == null) {
+            blockFactory = new BlockFactory(getRskSystemProperties().getBlockchainConfig());
+        }
+
+        return blockFactory;
+    }
+
     public TransactionPool getTransactionPool() {
         if (transactionPool == null) {
+            RskSystemProperties rskSystemProperties = getRskSystemProperties();
             transactionPool = new TransactionPoolImpl(
+                    rskSystemProperties,
+                    getRepository(),
                     getBlockStore(),
                     getReceiptStore(),
-                    getCompositeEthereumListener(),
+                    getBlockFactory(),
                     getProgramInvokeFactory(),
-                    getRepository(),
-                    getRskSystemProperties()
+                    getCompositeEthereumListener(),
+                    rskSystemProperties.txOutdatedThreshold(),
```

Comment:
I would hide these parameters inside TransactionPool constructor until rskSystemProperties is no longer a dependency for the class.

Reply:
I think this should be fixed along with `TransactionExecutor`, but not as part of this PR